daemon/graphdriver/aufs/aufs.go

```diff
@@ -272,16 +290,44 @@ func (a *Driver) Remove(id string) error {
 	if !exists {
 		mountpoint = a.getMountpoint(id)
 	}
-	if err := a.unmount(mountpoint); err != nil {
-		// no need to return here, we can still try to remove since the `Rename` will fail below if still mounted
-		logrus.Debugf("aufs: error while unmounting %s: %v", mountpoint, err)
+
+	var retries int
+	for {
+		mounted, err := a.mounted(mountpoint)
```

Comment:
EINVAL is returned for both "path is not mounted" as well as "path has other issues such as not a valid path". mounted() checks to see if mountpoint has a valid aufs magic and hence slightly better to use.